File: 881. Boats to Save People/Solution.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        n, count = len(people), 0
        left, right = 0, n - 1
        p = sorted(people)
        while left <= right:
            if p[left] + p[right] <= limit: left += 1
            right -= 1
            count += 1
        return count

File: 881. Boats to Save People/Solution.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        Arrays.sort(people);
        int left = 0, n = people.length, right = n - 1, count = 0;
        while (left <= right) {
            if (people[left] + people[right] <= limit) left++;
            right--;
            count++;
        }
        return count;
    }
}